x/posts/keeper/subspaces_hooks.go

```diff
@@ -0,0 +1,68 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/desmos-labs/desmos/v3/x/posts/types"
+
+	subspacestypes "github.com/desmos-labs/desmos/v3/x/subspaces/types"
+)
+
+// Hooks represents a wrapper struct
+type Hooks struct {
+	k Keeper
+}
+
+var _ subspacestypes.SubspacesHooks = Hooks{}
+
+// Hooks creates new relationships hooks
+func (k Keeper) Hooks() Hooks { return Hooks{k} }
+
+// AfterSubspaceSaved implements subspacestypes.Hooks
+func (h Hooks) AfterSubspaceSaved(ctx sdk.Context, subspaceID uint64) {
+	// Create the initial post it
+	h.k.SetNextPostID(ctx, subspaceID, 1)
+}
+
+// AfterSubspaceDeleted implements subspacestypes.Hooks
+func (h Hooks) AfterSubspaceDeleted(ctx sdk.Context, subspaceID uint64) {
+	// Delete the post id key
+	h.k.DeleteNextPostID(ctx, subspaceID)
+
+	// Delete all the posts
+	h.k.IterateSubspacePosts(ctx, subspaceID, func(_ int64, post types.Post) (stop bool) {
+		h.k.DeletePost(ctx, post.SubspaceID, post.ID)
+
+		// Emit an event
+		ctx.EventManager().EmitEvent(
+			sdk.NewEvent(
+				types.EventTypeDeletePost,
+				sdk.NewAttribute(types.AttributeKeySubspaceID, fmt.Sprintf("%d", post.SubspaceID)),
+				sdk.NewAttribute(types.AttributeKeyPostID, fmt.Sprintf("%d", post.ID)),
+			),
+		)
```

Comment:
It will create a tons of indexes for a block if there are millions of posts in a subspace. Can we simplify the event to be a general one like `EventTypeDeleteAllSubspacePost`?

Reply:
I've deleted it. The event `EventTypeSubspaceDeleted` should be sufficient to allow clients to react and delete the associated posts as well